server/service/service_osquery.go

```diff
@@ -446,15 +446,27 @@ func (svc *Service) detailQueriesForHost(ctx context.Context, host *fleet.Host)
 }
 
 func (svc *Service) shouldUpdate(lastUpdated time.Time, interval time.Duration, hostID uint) bool {
-	var jitter time.Duration
-	if svc.config.Osquery.MaxJitterPercent > 0 {
-		maxJitter := int64(svc.config.Osquery.MaxJitterPercent) * int64(interval) / 100.0
-		jitter = time.Duration((int64(hostID) + svc.getJitterSeed()) % maxJitter)
-	}
+	jitter := jitterForHost(
+		svc.config.Osquery.MaxJitterPercent,
+		svc.getJitterSeed(),
+		interval,
+		hostID,
+	)
 	cutoff := svc.clock.Now().Add(-(interval + jitter))
 	return lastUpdated.Before(cutoff)
 }
 
+func jitterForHost(maxJitterPercent int, jitterSeed int64, interval time.Duration, hostID uint) time.Duration {
```

Comment:
Can we get a unit test for this? Perhaps just asserting that the values fall in some sensible range?